File: client/src/app/objects/tarif.dto.ts
import { AbstractDto } from './abstract.dto';


export class TarifDto extends AbstractDto {
    public origine: string;
    public destination: string;
    public appelClasse2: number;
    public classe2: number;
    public classe1: number;
    public commentaire: string;

    constructor(data?: any) {
        super(data);

        data = data || {};
        this.origine = data.origine;
        this.destination = data.destination;
        this.appelClasse2 = data.appelClasse2;
        this.classe2 = data.classe2;
        this.classe1 = data.classe1;
        this.commentaire = data.commentaire;
    }
}

File: serveur/src/main/java/partirentrain/dto/TarifDto.java
package partirentrain.dto;

import java.text.Normalizer;
import java.util.function.Function;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class TarifDto {
	private String origine;
	private String destination;
	private double appelClasse2;
	private double classe2;
	private double classe1;
	private String commentaire;
	
	public boolean isOrigine(String origine) {
		Function<String, String> normalize = myStr -> {
			myStr = Normalizer.normalize(myStr, Normalizer.Form.NFD);
			myStr = myStr.replaceAll("[\\p{InCombiningDiacriticalMarks}]", "");
			myStr = myStr.replaceAll("-", "");
			myStr = myStr.replaceAll(" ", "");
			myStr = myStr.toLowerCase();
			return myStr;
		};

		return normalize.apply(this.origine).equals(normalize.apply(origine));
	}
}
